pkg/manifests/nginx.go

```diff
@@ -444,6 +448,14 @@ func newNginxIngressControllerDeployment(conf *config.Config, ingressConfig *Ngi
 						Name:  "controller",
 						Image: path.Join(conf.Registry, "/oss/kubernetes/ingress/nginx-ingress-controller:"+ingressConfig.Version.tag),
 						Args:  deploymentArgs,
+						Lifecycle: &corev1.Lifecycle{
```

Comment:
I'm not sure about what the effect is of adding the preStop is and if/why it's necessary, but the other part LGTM.

Reply:
I read through the upstream code, issues, and PRs and it seems to do absolutely nothing. Removing

Reply:
waiting on removal + will approve after!